File: p-e/spec/problem2spec.js
describe("PE Problem 2: Even Fibonacci Numbers:", function() {
  var evenFibs = new EvenFibs()
  it("Class can produce a list of n Fibonacci Numbers", function() {
    expect(evenFibs.genFibs(10)).toEqual([1, 2, 3, 5, 8, 13, 21, 34, 55, 89]);
  });

  it("Class can produce a list of Fibonacci Numbers up to a limit", function(){
    expect(evenFibs.fibsUpToCap(200).pop()).toEqual(144);
  })

  it("Class can filter for only even Fibonacci Numbers", function(){
    expect(evenFibs.evenFibsUpToCap(200)).toEqual([2, 8, 34, 144])
  });
});

File: p-e-clj/spec/p_e_clj/problem2_spec.clj
((ns p-e-clj.problem2-spec
  (:require [speclj.core :refer :all]
            [p-e-clj.problem2 :refer :all]))

(describe "PE Problem 2: Even Fibonacci Numbers:"
  (it "'nFibs' can produce a list of n Fibonacci Numbers"
    (should= '(1 2 3 5 8 13 21 34 55 89)
      (nFibs 10)))
  (it "'fibsUpToLim' produces a list of Fibonacci Numbers up to a limit"
    (should= 144
      (fibsUpToLim 200)))
  (it "'evenFibsUpToLim produces a list of even Fibonacci Numbers up to a limit"
    (should= '(2 8 34 144)
      (evenFibsUpToLim 200))))
(run-specs))